src/shared/utils/windowPeriod/fromAST.test.ts

```diff
@@ -0,0 +1,285 @@
+          })
+          array.from(rows: [{x: v.windowPeriod}])
+          */
+      const ast = textFixtures['inner_vs_outer_query']
+      runTest(ast, true, outerWindowPeriod)
+    })
+
+    describe('detect use of v.windowPeriod in the inner lambda', () => {
+      /*
+          import "influxdata/influxdb/sample"
+          sample.data(set: "airSensor")
+              |> range(start: -12h, stop: now())
+              |> map(fn: (r) => {
+                  return ({ r with _value: "${v.windowPeriod}" })
+              })
+          */
+      const ast = textFixtures['inner_lambda']
+      runTest(ast, true, 120000)
+    })
+
+    describe('two timeRanges used. Should grab the proper one for windowPeriod.', () => {
+      /*
+          import "influxdata/influxdb/sample"
+          sample.data(set: "airSensor")
+              |> range(start: -12h, stop: now())
+              |> yield(name: "${v.windowPeriod}")
+          sample.data(set: "airSensor")
+              |> range(start: -15m, stop: now())
+          */
+      const ast = textFixtures['multiple_ranges']
+      runTest(ast, true, 120000)
+    })
+
+    describe("when encountering a range node, only add a v.windowPeriod to the global scope if it doesn't already exist", () => {
+      /*
+          import "influxdata/influxdb/sample"
+          v = {windowPeriod: 2444}
+          sample.data(set: "airSensor")
+              |> range(start: -12h, stop: now())
+              |> yield(name: "${v.windowPeriod}")
+          */
+      const ast = textFixtures['when_to_inject']
+      runTest(ast, false, 2444)
+    })
+  })
+
+  describe('Can handle aliasing:', () => {
+    describe('of the timeRange property', () => {
+      /*
+          import "influxdata/influxdb/sample"
+          splitPoint = -1m
+          sample.data(set: "airSensor")
+              |> range(start: -12h, stop: splitPoint)
+              |> yield(name: "${v.windowPeriod}")
+          */
+      const ast = textFixtures['alias_range']
+      runTest(ast, true, 119833)
+    })
+    describe('of the timeRange variable', () => {
+      /*
+          import "influxdata/influxdb/sample"
+          splitPoint = -1m
+          v = {timeRangeStop: splitPoint}
+          sample.data(set: "airSensor")
+              |> range(start: -12h, stop: v.timeRangeStop)
+              |> yield(name: "${v.windowPeriod}")
+          */
+      const ast = textFixtures['alias_range_variable']
+      runTest(ast, true, 119833)
+    })
+    describe('of the windowPeriod variable', () => {
+      /*
+          import "array"
+          import "influxdata/influxdb/sample"
+          split = 2444
+          chunk = split
+          v = {windowPeriod: chunk}
+          array.from(rows: [{x: v.windowPeriod}])
+          */
+      const ast = textFixtures['alias_window_variable']
+      runTest(ast, false, 2444)
+    })
+  })
+
+  // TODO: https://github.com/influxdata/ui/issues/4695
+  describe.skip('Can handle complex inner expressions:', () => {
```

Comment:
Explicitly stating the limitations of this submodule contract. What it cannot do, as defined in these skipped tests, and the issue ticket for the missing functionality.